examples/udp/transport.cpp

```diff
@@ -0,0 +1,180 @@
+// Simple demonstration of data transfer over raw UDP.
+
+#include "caf/net/datagram_transport.hpp"
+#include "caf/net/ip.hpp"
+#include "caf/net/middleman.hpp"
+#include "caf/net/socket_event_layer.hpp"
+#include "caf/net/udp_datagram_socket.hpp"
+
+#include "caf/actor_system.hpp"
+#include "caf/actor_system_config.hpp"
+#include "caf/caf_main.hpp"
+#include "caf/event_based_actor.hpp"
+#include "caf/ipv4_address.hpp"
+#include "caf/log/net.hpp"
+
+#include <csignal>
+#include <iostream>
+#include <memory>
+#include <utility>
+
+using namespace caf;
+using namespace caf::net;
+
+using namespace std::literals;
+
+namespace {
+
+std::atomic<bool> shutdown_flag;
+
+// Implementation of logger interface for our custom component "app".
+namespace app {
+
+constexpr std::string_view component = "app";
+
+template <class... Ts>
+void debug(caf::format_string_with_location fmt_str, Ts&&... args) {
+  caf::logger::log(caf::log::level::debug, component, fmt_str,
+                   std::forward<Ts>(args)...);
+}
+
+template <class... Ts>
+void info(caf::format_string_with_location fmt_str, Ts&&... args) {
+  caf::logger::log(caf::log::level::info, component, fmt_str,
+                   std::forward<Ts>(args)...);
+}
+
+template <class... Ts>
+void error(caf::format_string_with_location fmt_str, Ts&&... args) {
+  caf::logger::log(caf::log::level::error, component, fmt_str,
+                   std::forward<Ts>(args)...);
+}
+
+} // namespace app
+
+} // namespace
+
+// -- configuration setup ------------------------------------------------------
+
+struct config : actor_system_config {
+  config() {
+    opt_group{custom_options_, "global"}
+      .add<std::string>("address,a", "Address to send payload to")
+      .add<std::string>("port,p", "Port to send payload to")
+      .add<uint16_t>("listen,l", "Port to listen for incoming data");
+  }
+};
+
+// -- utility functions --------------------------------------------------------
+
+// Read lines from standard input. Run as detached actor.
+void reader(event_based_actor* self, actor sink, ip_endpoint dest) {
+  std::string line;
+  while (std::getline(std::cin, line))
+    self->send(sink, dest, line);
+  // Stop the application when reading EOF.
+  shutdown_flag = true;
+}
+
+// Print lines to standard output.
+behavior printer(event_based_actor* self) {
+  return {[](ip_endpoint ep, std::string message) {
+            app::info("caf", "received {} bytes from {}:{}", message.size(),
+                      ep.address(), ep.port());
+            std::cout << message << std::endl;
+          },
+          [self](const message&) {
+            app::error("caf", "received unexpected message");
+            self->quit();
+          }};
```

Comment:
We always use a trailing comma. For better Git diffs, but also to get `clang-format` to do a better job formatting.

```suggestion
          },};
```